server/sonar-server/src/main/java/org/sonar/server/qualitygate/ws/DestroyAction.java

```diff
@@ -50,8 +54,16 @@ public void define(WebService.NewController controller) {
 
   @Override
   public void handle(Request request, Response response) {
-    qualityGates.delete(QualityGatesWs.parseId(request, QualityGatesWsParameters.PARAM_ID));
-    response.noContent();
+    long qualityGateId = request.mandatoryParamAsLong(QualityGatesWsParameters.PARAM_ID);
+    try (DbSession dbSession = dbClient.openSession(false)) {
+
+      QualityGateDto qualityGateDto = dbClient.qualityGateDao().selectById(dbSession, qualityGateId);
```

Comment:
Please replace this by QualityGateFinder.getById

Reply:
done